src/test/java/zd/zero/waifu/motivator/plugin/personality/NegativeEmotionCoreTests.kt

```diff
@@ -0,0 +1,366 @@
+            )
+        }
+    }
+
+    @Test
+    fun deriveMoodShouldReturnNeverReturnFrustratedWhenProbabilityIsZero() {
+        val emotionCore = EmotionCore(
+            WaifuMotivatorState().apply {
+                eventsBeforeFrustration = 0
+                probabilityOfFrustration = 0
+            }
+        )
+
+        val motivationEvent = buildMotivationEvent(
+            MotivationEvents.TASK,
+            MotivationEventCategory.NEGATIVE
+        )
+
+        val negativeEmotions =
+            OTHER_NEGATIVE_EMOTIONS.toTypedArray()
+        repeat(42) { index ->
+            val deriveMood = emotionCore.deriveMood(
+                motivationEvent
+            )
+            Assertions.assertThat(
+                deriveMood
+            ).withFailMessage(
+                """At index #$index
+                    |$motivationEvent
+                    |did not create $OTHER_NEGATIVE_EMOTIONS but did $deriveMood
+                """.trimMargin()
+            ).isIn(
+                *negativeEmotions
+            )
+        }
+    }
+
+    @Test
+    fun `frustration should evolve into rage`() {
+        val mockRandom = mockk<Random>()
+        val emotionCore = EmotionCore(
+            WaifuMotivatorState().apply {
+                eventsBeforeFrustration = 1
+                probabilityOfFrustration = 100
+            },
+            mockRandom
+        )
+
+        every { mockRandom.nextInt(1, 100) } returns 50
+        every { mockRandom.nextInt(2) } returns 1
+
+        val frustrated = Mood.FRUSTRATED.toList()
+        val enraged = Mood.ENRAGED.toList()
+        listOf(
+            buildMotivationEvent(
+                MotivationEvents.TASK,
+                MotivationEventCategory.NEGATIVE
+            ) to OTHER_NEGATIVE_EMOTIONS[1].toList(),
+            buildMotivationEvent(
+                MotivationEvents.TEST,
+                MotivationEventCategory.NEGATIVE
+            ) to frustrated,
+            buildMotivationEvent(
+                MotivationEvents.TEST,
+                MotivationEventCategory.NEGATIVE
+            ) to enraged,
+            buildMotivationEvent(
+                MotivationEvents.TEST,
+                MotivationEventCategory.NEGATIVE
+            ) to enraged,
+            buildMotivationEvent(
+                MotivationEvents.TEST,
+                MotivationEventCategory.NEGATIVE
+            ) to enraged
+        ).forEachIndexed { index, arguments ->
+            val deriveMood = emotionCore.deriveMood(
+                arguments.first
+            )
+            Assertions.assertThat(
+                deriveMood
+            ).withFailMessage(
+                """At index #$index
+                    |${arguments.first}
+                    |did not create ${arguments.second} but did $deriveMood
+                """.trimMargin()
+            ).isIn(arguments.second)
+        }
+    }
+
+    @Test
+    fun `frustration should cool down when positive events happen`() {
```

Comment:
> Is there a downside to using this on the test functions?

Not really, I just didn't do it before because I had the IDE generate the test for me. I was too much of a lazy bastard to bother to switch to this format. Fun Fact: You can use emojis in the \`\`

> Would you even suggest to use this?

Sure! In future tests and any changes I have to do to existing ones, I can convert to this format.